src/goto-instrument/contracts/contracts.cpp

```diff
@@ -171,33 +223,58 @@ void code_contractst::check_apply_loop_contracts(
       new_tmp_symbol(
         clause.type(), loop_head->source_location(), mode, symbol_table)
         .symbol_expr();
-    havoc_code.add(goto_programt::make_decl(
+    generated_code.add(goto_programt::make_decl(
       old_decreases_var, loop_head->source_location()));
     old_decreases_vars.push_back(old_decreases_var);
 
     const auto new_decreases_var =
       new_tmp_symbol(
         clause.type(), loop_head->source_location(), mode, symbol_table)
         .symbol_expr();
-    havoc_code.add(goto_programt::make_decl(
+    generated_code.add(goto_programt::make_decl(
       new_decreases_var, loop_head->source_location()));
     new_decreases_vars.push_back(new_decreases_var);
   }
 
-  // non-deterministically skip the loop if it is a do-while loop
+  // TODO: Fix loop contract issues for do/while loops.
   if(!loop_head->is_goto())
   {
-    havoc_code.add(goto_programt::make_goto(
+    log.error() << "Loop contracts are currently unsupported on do/while loops."
+                << messaget::eom;
+    throw 0;
+
```

Comment:
Shall we add a test case and label it as `KNOWNBUG`? It'd make codecov happy and it'd document this in our regression.

Reply:
Done; thanks!